src/test/java/RestAssuredApiTest.java

```diff
@@ -0,0 +1,225 @@
+    String sessionToken;
+    String bookingId;
+
+    @DataProvider(name = "originalData")
+    public Object[][] createOriginalData() {
+        return new Object [][] {
+                {"John", "Smith", 123, true, "2024-01-01", "2025-01-10", "Test Value"}
+        };
+    }
+
+    @DataProvider(name = "patchUpdateData")
+    public Object[][] createPatchUpdateData() {
+        return new Object [][] {
+                {321}
+        };
+    }
+
+    @DataProvider(name = "putUpdateData")
+    public Object[][] createPutUpdateData() {
+        return new Object [][] {
+                {"Mary", "Adams", 321, true, "2024-01-01", "2025-01-10", "Breakfast"}
+        };
+    }
+
+    //Authorisation To The https://restful-booker.herokuapp.com/ API
+
+    @BeforeTest
+    public void beforeSuite() {
+        String authLogin = "admin";
+        String authPassword = "password123";
+        Response res = RestAssured.given()
+                .body(String.format("{\"username\":\"%s\",\"password\":\"%s\"}", authLogin, authPassword))
+                .contentType(ContentType.JSON)
+                .when()
+                .post(String.format("%s/auth", baseURI))
+                .then()
+                .assertThat()
+                .statusCode(200)
+                .extract().response();
+        this.sessionToken = res.jsonPath().getString("token");
+        System.out.println(String.format("Token is %s", sessionToken));
+    }
+
+    //GET Request - Getting All The Bookings Present In The System
+
+    @Test(priority = 0)
+    public void getAllBookingsTest() {
+
+        Response res = RestAssured.given()
+                .log().all()
+                .when()
+                .get(String.format("%s/booking", baseURI))
+                .then()
+                .assertThat()
+                .statusCode(200)
+                .extract().response();
+
+        List<Integer> bookingIdList = res.jsonPath().getList("bookingid");
+
+        Assert.assertFalse(bookingIdList.isEmpty());
+
+        System.out.println(bookingIdList.size());
+    }
+
+    //POST Request - Creating New Booking
+
+    @Test(priority = 1, dataProvider = "originalData")
+    public void createNewBookingTest(String firstName, String lastName, int totalPrice, boolean depositPaid,
+                                     String checkInDate, String checkOutDate, String additionalNeeds) {
+
+
+        String inputJson = String.join(System.lineSeparator(),
+                String.format("{\"firstname\" : \"%s\",", firstName),
+                          String.format("\"lastname\" : \"%s\",", lastName),
+                          String.format("\"totalprice\" : %d,", totalPrice),
+                          String.format("\"depositpaid\" : %b,", depositPaid),
+                          String.format("\"bookingdates\" : { \"checkin\" : \"%s\", \"checkout\" : \"%s\" },", checkInDate, checkOutDate),
+                          String.format("\"additionalneeds\" : \"%s\"}", additionalNeeds));
```

Comment:
You can create json request as described here: https://www.codingninjas.com/studio/library/rest-assured--creating-json-object-and-array-request-body